### PR TITLE
meijerint_indefinite(f, x): use x as integration variable without assumptions

### LINKED ISSUE #25786: Wrong result for a simple integral
`sympy.integrate` returns a wrong result for a simple integral of a Gaussian function. The following code
```python
import sympy as sp
x = sp.Symbol('x')
f = sp.exp(-0.8*x**2)
sp.integrate(f, (x, -3, -0.2))
```
returns $−0.447288590994492\sqrt{\pi}$, which has the correct magnitude, but the wrong sign. This was observed with version 1.12 of `sympy`.

Interestingly, if the constant in the exponential is defined as a `Rational`:
```python
f = sp.exp(-sp.Rational(8,10)*x**2)
```
then the result is correct. If we use a different constant, for instance
```python
f = sp.exp(-0.4*x**2)
```
the result is also correct.

### PR DESCRIPTION
**Pull request by maliesen**
#### References to other Issues or PRs

Fixes #25786

#### Brief description of what is fixed or changed

meijerint_indefinite(f, x): use x as integration variable without any further assumptions

#### Release Notes

* integrals
  * Fixed a bug with MeijerG. Formerly, e.g. `exp(-x**2) over x=-2 to x=0` incorrectly integrated to `-erf(2)*sqrt(pi)/2`.

### CODE REVIEW

File: sympy/integrals/meijerint.py [hunk 1]
```diff
@@ -1664,20 +1664,25 @@ def meijerint_indefinite(f, x):
     -cos(x)
     """
     f = sympify(f)
+    x_ = x
+    d = Dummy('x')
+    f = f.subs(x, d)
+    x = d
```
Comment (asmeurer) [inline sympy/integrals/meijerint.py:1670]:
```suggestion
    x_, x = x, Dummy('x')
    f = f.subs(_x, x)
```

File: sympy/integrals/meijerint.py [hunk 2]
```diff
@@ -1664,20 +1664,25 @@ def meijerint_indefinite(f, x):
...
+
     results = []
     for a in sorted(_find_splitting_points(f, x) | {S.Zero}, key=default_sort_key):
         res = _meijerint_indefinite_1(f.subs(x, x + a), x)
         if not res:
             continue
-        res = res.subs(x, x - a)
+        res = res.subs(x, x - a).subs(x, x_)
```
Comment (asmeurer) [inline sympy/integrals/meijerint.py:1677]:
```suggestion
        res = res.subs(x, x_ - a)
```

File: sympy/integrals/meijerint.py (hunk 1 above)
Comment (asmeurer) [reply sympy/integrals/meijerint.py:1670]: And add a comment here why we are doing this. Is the point to ignore assumptions?

**Comment by asmeurer** [comment]
How does assumptions break the code here? We should see if whatever isn't working with symbols with assumptions can be fixed directly.

**Comment by maliesen** [comment]
@asmeurer  Thanks for your review. I think you are right with your last point about analyzing whether what isn't working with symbols with assumptions can be fixed directly. For this bug there might be several issues involved. I'm new to sympy so some things might work differently than I would have guessed. One thing I don't get is the [lookup table](https://docs.sympy.org/latest/modules/integrals/g-functions.html#implemented-g-function-formulae) . E.g., it states 

> \begin{split}\sin{\left(z^{q} p \right)} = \sqrt{\pi} {G_{0, 2}^{1, 0}\left(\begin{matrix}  &  \\\frac{1}{2} & 0 \end{matrix} \middle| {\frac{z^{2 q} p^{2}}{4}} \right)}\end{split}

Now if you plug in `p=q=1` the right hand side is a function of `z**2` thus being insensitive to the sign of `z` while `sin(z)` is clearly not. Do I perhaps read it wrong? What is the domain of the variables involved? (`p,q,z`)

**Comment by asmeurer** [comment]
Yes, the lookup table rules should have conditions. If I'm not mistaken, that formula is here https://github.com/sympy/sympy/blob/master/sympy/integrals/meijerint.py#L171-L172. The condition should be added as the `cond` argument of `add()` https://github.com/sympy/sympy/blob/master/sympy/integrals/meijerint.py#L96. So unless I'm mistaken, that condition is missing from the code. 

However, I tried adding it and it doesn't affect this integral. It might affect others though. It would be a good idea to double check this and fix it and any others that are missing conditions.

Here's the `SYMPY_DEBUG=True` output for this integral

```
Trying to compute the indefinite integral of exp(-_Dummy_36**2) wrt _Dummy_36
 could rewrite: (1, 1, [(1, 0, meijerg(((), ()), ((0,), ()), _Dummy_36**2))], True)
Try to expand Meijer G function corresponding to G_Function((1,), (), (1/2,), (0,))
  Could not reduce order.
  Could not find a direct formula. Trying Slater's theorem.
Trying to expand hypergeometric function Hyper_Function((1/2,), (3/2,))
  Could not reduce order.
  Found an origin: sqrt(_z)*I*sqrt(pi)*erf(sqrt(_z)*I) + exp(_z) Hyper_Function((-1/2,), (1/2,))
```

Perhaps one of the erf lookup rules is also missing a condition.

**Comment by oscarbenjamin** [comment]
The slow test that fails with this PR is:
```python
In [7]: from sympy import *
   ...: from sympy.abc import x, y
   ...: a, b = symbols('a b', positive=True)
   ...: betadist = x**(a - 1)*(-x + 1)**(b - 1)*gamma(a + b)/(gamma(a)*gamma(b))
   ...: f = x**y*betadist

In [8]: f
Out[8]: 
 y  a - 1        b - 1         
x ⋅x     ⋅(1 - x)     ⋅Γ(a + b)
───────────────────────────────
           Γ(a)⋅Γ(b)           

In [9]: integrate(f, (x, 0, 1), meijerg=True)
Out[9]: 
   a + y                    ┌─  ⎛a + y, 1 - b │  ⎞                      ┌─  ⎛a + y, 1 - b │  ⎞
  0     ⋅Γ(a + b)⋅Γ(a + y)⋅ ├─  ⎜             │ 0⎟   Γ(a + b)⋅Γ(a + y)⋅ ├─  ⎜             │ 1⎟
                           2╵ 1 ⎝ a + y + 1   │  ⎠                     2╵ 1 ⎝ a + y + 1   │  ⎠
- ──────────────────────────────────────────────── + ─────────────────────────────────────────
               Γ(a)⋅Γ(b)⋅Γ(a + y + 1)                         Γ(a)⋅Γ(b)⋅Γ(a + y + 1)  
```
Note that we have `0**(a+y)`. I suspect that the positive=True assumptions somewhere would cause that to have simplified to zero.

With master the integral gives only the second term:
```python
In [3]: integrate(f, (x, 0, 1), meijerg=True)
Out[3]: 
                   ┌─  ⎛a + y, 1 - b │  ⎞
Γ(a + b)⋅Γ(a + y)⋅ ├─  ⎜             │ 1⎟
                  2╵ 1 ⎝ a + y + 1   │  ⎠
─────────────────────────────────────────
         Γ(a)⋅Γ(b)⋅Γ(a + y + 1) 
```
In this case it simplifies:
```python
In [4]: simplify(integrate(f, (x, 0, 1), meijerg=True))
Out[4]: 
Γ(a + b)⋅Γ(a + y)
─────────────────
Γ(a)⋅Γ(a + b + y)
```
I don't know whether this is generally correct for all possible values of `a`, `y` and `b` and in particular for negative `y`. It looks like this is calculating `E[x^y]` where x follows the beta distribution:
https://en.wikipedia.org/wiki/Beta_distribution#Higher_moments

**Comment by oscarbenjamin** [comment]
> How does assumptions break the code here? We should see if whatever isn't working with symbols with assumptions can be fixed directly.

One possibility is that although an integral might be along the real line the meijerint method computes the integral indirectly using intermediate complex valued expressions. A simplification like `sqrt(x**2) -> x` (for positive x) might be invalid for the intermediate calculations leading to a wrong result overall.

I am not sure I fully understand all steps n the meijerint method but I think that it is fundamentally incorrect to set assumptions on the integration variable for meijerint and also possible for other methods as well.

**Pull request closed without merging**